tests/melt_transport_adaptive.cc

```diff
@@ -0,0 +1,404 @@
+#include <aspect/material_model/interface.h>
+#include <aspect/velocity_boundary_conditions/interface.h>
+#include <aspect/fluid_pressure_boundary_conditions/interface.h>
+#include <aspect/simulator_access.h>
+#include <aspect/global.h>
+#include <aspect/mesh_refinement/interface.h>
+#include <aspect/melt.h>
+
+#include <deal.II/dofs/dof_tools.h>
+#include <deal.II/numerics/data_out.h>
+#include <deal.II/base/quadrature_lib.h>
+#include <deal.II/base/function_lib.h>
+#include <deal.II/numerics/error_estimator.h>
+#include <deal.II/numerics/vector_tools.h>
+
+
+namespace aspect
+{
+  template <int dim>
+  class MeltVelocityRefinement : public MeshRefinement::Interface<dim>,
```

Comment:
Is it worth adding this as an actual mesh refinement criterion in a separate PR later?